kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/LoadBalancerClientTest.java

```diff
@@ -403,52 +389,47 @@ public void run() {
                                 "  <kie-server-info>\n" +
                                 "    <version>background scan</version>\n" +
                                 "  </kie-server-info>\n" +
-                                "</response>"))
-                .willSetStateTo("Good Req 1"));
+                                "</response>")));
 
-        // Request to indicate server is back online
-        wireMockServer1.stubFor(get(urlEqualTo("/state"))
-                .withHeader("Accept", equalTo("application/xml"))
-                .inScenario("Timeout Fails followed by Scan Success")
-                .whenScenarioStateIs("Good Req 1")
-                .willReturn(aResponse()
-                        .withStatus(200)
-                        .withHeader("Content-Type", "application/xml")
-                        .withBody("<response type=\"SUCCESS\" msg=\"Kie Server state\">\n" +
-                                "  <kie-server-state-info>\n" +
-                                "    <version>1c</version>\n" +
-                                "  </kie-server-state-info>\n" +
-                                "</response>"))
-                .willSetStateTo("Good Req 2"));
 
         // Kickoff first scenario
         int threadCount=2;
         logger.debug("Starting 2 Threads");
         CountDownLatch startLatch = new CountDownLatch(1);
         CountDownLatch stopLatch = new CountDownLatch(threadCount);
-        for (int i=1; i<= threadCount; i++) {
-            new SendKieRequestThread(i, startLatch, stopLatch, client).start();
+        List<SendKieRequestThread> threads = new ArrayList<>();
+        for (int i = 1; i <= threadCount; i++) {
+            threads.add(new SendKieRequestThread(i, startLatch, stopLatch, client));
         }
+        threads.forEach(SendKieRequestThread::start);
+
         // Threads will be waiting to proceed, so let them off.
         startLatch.countDown();
 
         // We expect the threads to complete within 7 seconds
         stopLatch.await(7, TimeUnit.SECONDS);
         logger.debug("\nEnd of Threads - ");
-
-        // Need to sleep to ensure background thread completes.
-        logger.debug("\nSleeping for 3 seconds - ");
-        Thread.sleep(3000);
+        threads.forEach(thread -> {
+            try {
+                thread.join();
+            } catch (InterruptedException e) {
+                logger.debug("Interrupted", e);
+            }
+        });
 
         // Expect to now have server:/state incorrectly retained in failedEndpoints
         List<String> availableList = ((AbstractKieServicesClientImpl)client).getLoadBalancer().getAvailableEndpoints();
         availableList.forEach(item -> logger.debug("Available Endpoint : [" + item + "]"));
         List<String> failedList = ((AbstractKieServicesClientImpl)client).getLoadBalancer().getFailedEndpoints();
         failedList.forEach(item -> logger.debug("Failed Endpoint : [" + item + "]"));
```

Comment:
It would be good to check/assert the availableEndpoints along with the failedEndpoints here

Reply:
done